pkg/datasource/database/ddl.go

```diff
@@ -0,0 +1,89 @@
+package database
+
+import "fmt"
+
+const ddlLoaded = `SELECT EXISTS (SELECT 1 FROM pg_namespace WHERE nspname = $1);`
+
+const createDeleteSchemaFunctions = `
```

Comment:
The intention was more to move the database structure creation logic into the code, rather than rely on the deployment to do so. I find this simplifies the deployment since:
1. in test deployment we have less control over the database since its deployment is done through geco
2. in production deployment the plugin is loaded by geco and not as an independent runtime, and I wanted to avoid the need of having to run devops stuff from geco

And yes one of the other objective was to support multiple data sources of the same type.

> why aren't you creating the explore query and saved cohorts table in the correspondent schema

It is, through a separate statement `ddlStatement`.

Reply:
Probably it's because of my not great proficiency in sql, but in the `ddlstatement` I don't see specified the schema in which the two tables are created. Aren't the table created in the default schema (i.e., public) in this case?

Reply:
So for this, the connection string passed to the postgresql driver contains the schema ([here](https://github.com/ldsec/geco-i2b2-data-source/pull/6/files/01ac9e78e02961e0c74211918261cb22f9c55522#diff-00218109bdc70c81c42d1fdf852be9d8078b58540668c56f63800f6c9f485f8dR14)), i.e. if the schema is not specified in the SQL query, it will default to the one passed to the driver. In this case, the name of the schema is configurable.